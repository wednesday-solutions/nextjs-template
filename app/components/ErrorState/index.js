/**
 *
 * ErrorState
 *
 */

import React from 'react';
import get from 'lodash/get';
import { compose } from 'redux';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import T from '@components/Text';
import { CustomCard } from '../styled/repos';

const ErrorState = (props) => {
  const { intl, reposError, loading, reposData } = props;
  const getRepoError = () => {
    if (reposError) {
      return reposError;
    } else if (!get(reposData, 'totalCount', 0)) {
      return 'respo_search_default';
    }
    return null;
  };

  const renderErrorCard = (repoError) => {
    return (
      !loading &&
      repoError && (
        <CustomCard
          color={reposError ? 'red' : 'grey'}
          title={intl.formatMessage({ id: 'repo_list' })}
          data-testid="error-state"
        >
          <T id={repoError} />
        </CustomCard>
      )
    );
  };

  const repoError = getRepoError();
  return renderErrorCard(repoError);
};

ErrorState.propTypes = {
  intl: PropTypes.any,
  loading: PropTypes.bool.isRequired,
  reposData: PropTypes.arrayOf(
    PropTypes.shape({
      totalCount: PropTypes.number,
      incompleteResults: PropTypes.bool,
      items: PropTypes.array
    })
  ),
  reposError: PropTypes.object,
  repoName: PropTypes.string,
  recommendations: PropTypes.arrayOf(
    PropTypes.shape({ id: PropTypes.number.isRequired, name: PropTypes.string.isRequired })
  )
};

export default compose(injectIntl)(ErrorState);
