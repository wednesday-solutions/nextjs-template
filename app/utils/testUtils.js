import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../configureStore';
import colors from '@themes/colors';
import { translationMessages, DEFAULT_LOCALE } from '../i18n';

export const getComponentStyles = (Component, props = {}) => {
  renderProvider(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);

  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = (children) => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={colors}>{children}</ThemeProvider>
    </IntlProvider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const renderStoreProvider = (children) => {
  const store = configureStore({});
  return render(
    <Provider store={store}>
      <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
        <ThemeProvider theme={colors}>{children}</ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};
