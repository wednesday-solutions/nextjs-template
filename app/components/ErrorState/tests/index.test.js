/**
 *
 * Tests for ErrorState
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderProvider } from '@utils/testUtils'
import ErrorState from '../index'

describe('<ErrorState />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<ErrorState />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 ErrorState component', () => {
    const { getAllByTestId } = renderProvider(
      <ErrorState repoError='failure' />
    )
    expect(getAllByTestId('error-state').length).toBe(1)
  })
})
