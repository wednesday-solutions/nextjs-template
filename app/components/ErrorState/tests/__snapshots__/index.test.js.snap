// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorState /> should render and match the snapshot 1`] = `
.emotion-0.emotion-0 {
  margin: 20px 0;
  color: grey;
  color: grey;
}

.emotion-2 {
  white-space: pre-line;
}

<body>
  <div>
    <div
      class="ant-card ant-card-bordered emotion-0 emotion-1"
      color="grey"
      data-testid="error-state"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Repository List
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <span
          class="emotion-2"
          data-testid="text"
        >
          Search for a repository by entering it's name in the search box
        </span>
      </div>
    </div>
  </div>
</body>
`;
